Проектная работа №2 (спринт 4) тема "Применяем дженерики"

DoNatPanic opened the pull request.

sbuvaka-fhl committed:
Add dev branch for Y.Practicum

DoNatPanic committed:
Refactor code: replace ListDiffCallback in each adapter with base class

В файлах:
ContactAdapter,
FilterContactTypeAdapter и
SortTypeAdapter,

которые предназначены для вывода элементов списка контактов, убрали дублирующийся класс ListDiffCallback и заменили его на базовый дженерик-класс;
обобщенный тип этого класса ограничен реализацией интерфейса ListDiffInterface.

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/ListDiffInterface.java:
```diff
@@ -0,0 +1,7 @@
+
+public interface ListDiffInterface<T> {
+
+    boolean theSameAs(T t);
```
Лучше использовать такой нейминг, чтобы по названию было понятно, что параметр означает. Например, anotherItem вместо t

DoNatPanic replied:
Спасибо, буду иметь ввиду

DoNatPanic committed:
Fix problems #1

В классе BaseListDiffCallback добавлена реализация метода getChangePayload
В классах FilterContactTypeUi и SortTypeUI исправлена реализация метода theSameAs

DoNatPanic commented:
В последнем коммите исправлены критические замечания
